### ISSUE TITLE
Preflight requests handled by `@Filter`s blocked by CorsFilter

### ISSUE THREAD

**Issue by ThomasIommi** [micronaut-projects/micronaut-core#11375]
Hello!

As mentioned in issue #7934, it seems that the router might not recognize paths defined using patterns like those used by `@Filter`s. 
I'm afraid that checking for known paths reintroduce the same issue, causing all requests to a proxy filter to be blocked by the CORS filter with a 403 Forbidden error.

This situation seems to prevent us from implementing solutions suggested in the official documentation, such as the one found here: https://docs.micronaut.io/latest/guide/#proxyClient

Here's a snippet of the code in question:

```java
@Nullable
private boolean validatePreflightRequest(@NonNull HttpRequest<?> request,
                                         @NonNull CorsOriginConfiguration config) {
    ...
    List<HttpMethod> availableHttpMethods = router.findAny(request).stream().map(UriRouteMatch::getHttpMethod).toList();
    if (availableHttpMethods.stream().noneMatch(method -> method.equals(methodToMatch))) {
        return false;
    }    
    ...
    return true;
}
```

_Originally posted by @ThomasIommi in https://github.com/micronaut-projects/micronaut-core/issues/11097#issuecomment-2495529232_

**Comment by yawkat** [comment]
The core issue is that filters are not recognized by the CORS filter, and the solution will be the same (if there is one), so I'm closing this as duplicate of #7934.
